src/core/toga/widgets/canvas.py

```diff
@@ -313,3 +313,31 @@ def reset_transform(self):
 
         """
         self._impl.reset_transform()
+
+    # Text
+
+    def write_text(self, text, x=0, y=0):
+        """Writes a given text
+
+        Writes a given text at the given (x,y) position.
```

Comment:
This API requires an implicit font declaration - is that the best option?

Reply:
I'm not sure if it is the best option either. My strategy was to set the font on the Canvas widget. Alternatively I could have explicitly passed it through the write_text method or set it as part of the widget style. I'm not sure what your vision is for how this will work with travertine, so if you have a preference I'm happy to make changes to make it more explicit. I could also check if the method has a passed in font, and if not look for a font on the widget.

Reply:
Sorry - a clarification: my reference to Travertino was about the definition itself - travertino has a built-in, cross-platform definition for color, and it will (eventually) have a similar definition for font. Any APIs will eventually need to support those types as inputs, cast into platform-specific renditions on the platform backend.

As for the API on canvas itself; while it makes sense for a "Label" widget to have a "set font" API (because there's only one label that can be rendered), I'm not sure the same is true of Canvas. It's not hard to imagine uses of a canvas that would involve several fonts, in multiple sizes. Having "set font" to provide a *default* font makes some sense, but anything individual drawing operation should have the ability to override that default.

It's similar to color - the Canvas doesn't have "*the* foreground color", there's just the foreground color we're currently drawing in, which may fall back to a useful default if unspecified. Of course, that doesn't mean every individual "draw line" operation needs to have a color argument (although that would make a certain amount of sense), but there definitely needs to be an obvious "set the font/color for the current drawing context" operation.

Reply:
That makes sense. I agree that "set font" on the widget would be more like a default, and then each drawing operation can pass in a unique one. I pushed a new version, please let me know if this was along the lines of what you were thinking.